test/cmdlineTests/debug_info_in_yul_and_evm_asm_print_all/output

```diff
@@ -201,16 +201,14 @@ object "C_6" {
         code {
             {
                 /// @src 0:60:101  "contract C {..."
-                let _1 := memoryguard(0x80)
-                mstore(64, _1)
                 if iszero(lt(calldatasize(), 4))
                 {
-                    let _2 := 0
-                    if eq(0x26121ff0, shr(224, calldataload(_2)))
+                    let _1 := 0
+                    if eq(0x26121ff0, shr(224, calldataload(_1)))
                     {
-                        if callvalue() { revert(_2, _2) }
-                        if slt(add(calldatasize(), not(3)), _2) { revert(_2, _2) }
-                        return(_1, _2)
+                        if callvalue() { revert(_1, _1) }
+                        if slt(add(calldatasize(), not(3)), _1) { revert(_1, _1) }
+                        return(memoryguard(0x80), _1)
```

Comment:
We could consider to do ``return(x,0) -> return(0,0)`` and ``revert(x,0) -> revert(0,0)`` in the evmasm optimizer, if we don't do that already.
Or on the Yul level ``return(x,0) -> pop(x) return(0,0)`` (we can't just remove the memoryguard)...

Reply:
Sounds good!

Reply:
Although we should rather do that in the yul optimizer shouldn't we?

Reply:
Yeah, I first thought we couldn't, but ``return(x,0) -> pop(x) return(0,0)`` we *can* do in the yul optimizer.
``pop(memoryguard(...))`` would still stay - we could resolve the memoryguard and try to remove it *after* stack-to-memory, but then we have to be careful not to do anything that really affects stack layout anymore... so since ``pop <constant>`` will just be removed in evmasm, I'd say just doing ``return(x,0) -> pop(x) return(0,0)`` in Yul is good enough....